### DISCUSSION TITLE
Data Replication & Consensus Protocol Support

### DISCUSSION THREAD

**Discussion by pinghe** [agnesoft/agdb#1186] [General]
## Feature Inquiry: Data Replication & Consensus Protocol Support

**Feature Description:**
I am interested in knowing if the following feature is currently supported in the project:

- **Data Replication & Consensus Protocol:** Allow replication by connecting several database nodes together with a RAFT protocol.

**Purpose:**
This feature is essential for ensuring data consistency and fault tolerance across distributed database systems.

**Questions:**
1. Is the RAFT protocol for data replication currently implemented in the project?
2. If so, could you provide documentation or examples on how to set it up?
3. If not, is there a roadmap or timeline for when this feature might be considered or implemented?

I look forward to your response regarding the availability and support of this feature.

**Comment by michaelvlach** [comment]
H, and yes, RAFT is on the [roadmap](https://github.com/agnesoft/agdb?tab=readme-ov-file#roadmap)! I have started on it some time back but other things took priority. I plan to get back to it after the PHP driver so I hope to get it released as the major feature in 0.9.0. As for the timelines I cannot promise any specific date but realistically I expect it to take me months as I am not able to work on `agdb` full time yet.

**Comment by michaelvlach** [comment]
Going through old threads RAFT is supported since 0.10.0 released in January 2025.
